core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -107,7 +107,9 @@ public String toString() {
      *
      * @param hasPrivateKey Whether it is a path to a private key or not
      * @param list List of children in the path
+     * @deprecated Use {@link HDPath#of(Prefix, List)} or another static constructor
      */
+    @Deprecated
```

Comment:
Shouldn't we make the other constructor the canonical one, before deprecating this one? Otherwise, this constructor will still be used by us – despite the deprecation.

Reply:
There are even more usages, imho we should all migrate them in the same commit.

Reply:
It's deprecated for being made private, not for removal.

(However, we may have to remove it without deprecation if we implement HDFullPath and HDPartialPath subclasses)

Reply:
So this will stay the canonical constructor?

Reply:
Probably, because internally we are using a primitive `boolean` for memory reasons.

Reply:
Ok. Still I think the references from tests should be migrated to some public non-deprecated constructor, if that's possible. Otherweise we can't make it private anyway, would need to change it to package-private.

Reply:
I thought I had done that, will double check.

Reply:
> So this will stay the canonical constructor?

Yes, it stays the canonical constructor in PR #3721, but it is made private. (which means unless we change something there will be no deprecation)

Reply:
>> I think the references from tests should be migrated to some public non-deprecated constructor

> I thought I had done that, will double check.

I double-checked and I don't see any.

I'm going to "Resolve this conversation", but we can continue it on PR #3722 if you like.